AI_ML/OPENAI/WHISPER_SPEECH_TO_TEXT/WHISPER_SPEECH_TO_TEXT.py

```diff
@@ -0,0 +1,68 @@
+from flojoy import flojoy, DataContainer, DataFrame as FlojoyDataFrame, Bytes
+from typing import List, Optional
+import openai
+import pandas as pd
+import os
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+
+ACCEPTED_AUDIO_FORMATS = ["mp3", "wav"]
+
+
+@flojoy
+def WHISPER_SPEECH_TO_TEXT(
+    default: Optional[Bytes] = None, 
+    file_path: Optional[str] = None
+) -> FlojoyDataFrame:
+    """
+    This node uses OpenAI whisper transcription model to convert audio to text.
+    The audio can be provided as a file path or as bytes from a previous node.
+    The previous node value has priority over the file path.
+
+    Parameters:
+    -----------
+    - file_path: string
+        Path to the audio file to be transcribed. Only mp3 format is supported.
+    """
+    openai.api_key = os.environ.get("OPENAI_API_KEY")
+    if default and isinstance(getattr(default, 'bytes'), bytes):
+        # if bytes exists create temp file and use it
+        with NamedTemporaryFile(suffix='.mp3') as f:
+            f.write(default.bytes)
+            f.seek(0)
+            openai.api_key = os.environ.get("OPENAI_API_KEY")
+            model = 'whisper-1'
+            transcript = openai.Audio.translate(model, f)
+            transcript_text = transcript.get('text')
+            transcription_df = pd.DataFrame(data={
+                'text': [transcript_text]
+            })
+
+            return DataContainer(
+                type='dataframe',
+                m=transcription_df
+            )
```

Comment:
much of this code is duplicated below. I suggest:
* changing the logic and creating the file outside of the context manager and setting the `file_path` so it can be picked up below, and then making the `if file_path is None` into `elif file_path is None` and letting the rest of the code do the work


something like
```
if default and isinstance(getattr(default, "bytes"), bytes):
    f = NamedTemporaryFile(suffix=".mp3")
    f.write(default.bytes)
    f.seek(0)
    file_path = f.name
elif file_path is None:
    raise ValueError("file_path parameter is missing!")
# the same code you have now
```

This will remove duplication and make the code easier to read